### PR TITLE
introduce NewWildcardIdentitiesRoundTripper

### PR DESCRIPTION
**Pull request by p0lyn0mial**
## Summary
NewWildcardIdentitiesRoundTripper creates a HTTP RoundTripper that injected resource identities for
individual group or group resources. Each group or resource is coming from one APIExport whose
names are passed in as a map.

The returned resolve function will get the APIExports and extract the identities.
The resolve func might return an error if the APIExport is not found or for other reasons. Only
after it succeeds a client using the returned RoundTripper can use the group and group resources
with identities.

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/server/bootstrap/identity.go [hunk 1]
```diff
@@ -84,7 +84,23 @@ func NewConfigWithWildcardIdentities(config *rest.Config,
...
 
+// NewWildcardIdentitiesRoundTripper creates a HTTP RoundTripper that injected resource identities for
+// individual group or group resources. Each group or resource is coming from one APIExport whose
+// names are passed in as a map.
+//
+// The returned resolve function will get the APIExports and extract the identities.
+// The resolve func might return an error if the APIExport is not found or for other reason. Only
+// after it succeeds a client using the returned RoundTripper can use the group and group resources
+// with identities.
+func NewWildcardIdentitiesRoundTripper(groupExportNames map[string]string,
```
Comment (sttts) [inline pkg/server/bootstrap/identity.go:101]: it's not really the roundtripper, but a roundtripper wrapper, right?

Comment (p0lyn0mial) [reply pkg/server/bootstrap/identity.go:101]: it returns a RoundTripper and a function that allows for wrapping

Comment (p0lyn0mial) [reply pkg/server/bootstrap/identity.go]: updated the function name and changed the gdoc.

**Comment by p0lyn0mial** [comment]
/close

pulled into #1572

**Pull request closed without merging**